cumulusci/core/tests/test_flows.py

```diff
@@ -111,6 +128,181 @@ def test_pass_around_values(self):
+            'tasks': {
+                1: {'task': 'pass_name'},
+                2: {'task': 'None'},
+            }
+        })
+
+        flow = BaseFlow(
+            self.project_config,
+            flow_config,
+            self.org_config,
+            skip=['name_response'],
+        )
+
+        # run the flow
+        flow()
+
+        # the number of tasks in the flow should be 1 instead of 2
+        self.assertEquals(1, len(flow.task_results))
+
+    def test_find_task_by_name_no_tasks(self):
+        """ The _find_task_by_name method skips tasks that don't exist """
+
+        # instantiate a flow with two tasks
+        flow_config = FlowConfig({
+            'description': 'Run two tasks',
+        })
+
+        flow = BaseFlow(
+            self.project_config,
+            flow_config,
+            self.org_config,
+        )
+
+        self.assertEquals(None, flow._find_task_by_name('missing'))
+
+    def test_find_task_by_name_not_first(self):
+        """ The _find_task_by_name method skips tasks that don't exist """
+
+        # instantiate a flow with two tasks
+        flow_config = FlowConfig({
+            'description': 'Run two tasks',
+            'tasks': {
+                1: {'task': 'pass_name'},
+                2: {'task': 'name_response', 'options': {
+                    'response': '^^pass_name.name'
+                }},
+            }
+        })
+
+        flow = BaseFlow(
+            self.project_config,
+            flow_config,
+            self.org_config,
+        )
+
+        flow()
+
+        task = flow._find_task_by_name('name_response')
+        self.assertEquals(
+            'cumulusci.core.tests.test_flows._TaskResponseName',
+            task.task_config.class_path,
+        )
+
+    def test_render_task_config_empty_value(self):
+        """ The _render_task_config method skips option values of None """
+
+        # instantiate a flow with two tasks
+        flow_config = FlowConfig({
+            'description': 'Run a tasks',
+            'tasks': {
+                1: {'task': 'name_response',
+                    'options': {
+                        'response': None,
+                    },
+            
+                   },
+            },
+        })
+
+        flow = BaseFlow(
+            self.project_config,
+            flow_config,
+            self.org_config,
+        )
+
+        flow()
+
+        task = flow._find_task_by_name('name_response')
+        config = flow._render_task_config(task)
+        self.assertEquals(['Options:'], config)
+
+    def test_task_raises_exception_fail(self):
+        """ A flow aborts when a task raises an exception """
+
+        flow_config = FlowConfig({
+            'description': 'Run a task',
+            'tasks': {
+                1: {'task': 'raise_exception'},
+            }
+        })
+        flow = BaseFlow(self.project_config, flow_config, self.org_config)
+        self.assertRaises(Exception, flow)
```

Comment:
Shouldn't assertRaises be used in context manager? E.g. `with self.assertRaises(Exception):`

Reply:
The context manager is only needed if you want to check attributes on the exception that's raised.  self.assertRaises will call the second arg and assert that it raises an instance of the first arg.  I tried both ways and the traceback still gets printed out.